cmd/soroban-rpc/lib/preflight/src/preflight.rs

```diff
@@ -46,6 +46,12 @@ pub(crate) fn preflight_invoke_hf_op(
         .context("cannot create budget")?;
     let storage = Storage::with_recording_footprint(ledger_storage_rc.clone());
     let host = Host::with_storage_and_budget(storage, budget);
+    host.set_source_account(source_account.clone())
```

Comment:
The source account passed in here is either the operation source account or the transaction source account, right ? and that logic aligns with the core's implementation, right ? @sisuresh could you confirm that ?

Reply:
Yeah that aligns with what core passes to the host - https://github.com/stellar/stellar-core/blob/23f3cfb821d79f3958b615e0404d6f6ff446cdaa/src/transactions/OperationFrame.cpp#L220.